### PR TITLE
Numpy Pooling and ROI Pooling Large Dimension Checks

### PR DESCRIPTION
**Pull request by Zha0q1**
This PR adds large dimension checks to pooling and roi pooling. Those operators should not have large dimension use cases but we add in the checks just to be safe. Also I updated the large tensor tests for the two operators

### CODE REVIEW

File: src/operator/nn/pooling.cc [hunk 1]
```diff
@@ -117,6 +117,11 @@ static bool PoolingShape(const nnvm::NodeAttrs &attrs,
       << " Or 4D in (batch, channel, y, x) "
       << " Or 5D in (batch, channel, d, y, x)";
 
+  for (int i=0; i<dshape.ndim(); i++) {
+    CHECK_LT(dshape[i], (int64_t{1} << 31) - 1) <<
```
Comment (wkcn) [inline src/operator/nn/pooling.cc:121]: We can replace `(int64_t{1} << 31) - 1` with `INT32_MAX`.

File: src/operator/roi_pooling-inl.h [hunk 2]
```diff
@@ -177,6 +177,11 @@ class ROIPoolingProp : public OperatorProperty {
     mxnet::TShape dshape = in_shape->at(roipool::kData);
     CHECK_EQ(dshape.ndim(), 4U) << "data should be a 4D tensor";
 
+    for (int i=0; i<dshape.ndim(); i++) {
+        CHECK_LT(dshape[i], (int64_t{1} << 31) - 1) <<
```
Comment (wkcn) [inline src/operator/roi_pooling-inl.h:181]: We can replace `(int64_t{1} << 31) - 1` with `INT32_MAX`.

File: tests/nightly/test_np_large_array.py [hunk 3]
```diff
@@ -1005,33 +1006,44 @@ def test_dlpack():
...
 def test_pooling():
-    A = np.ones((1, 2, INT_OVERFLOW))
-    A[0][0][2] = 100
+    def test_pooling_large_dim():
+        A = np.ones((1, 1, INT_OVERFLOW))
+        assertRaises(MXNetError, npx.pooling, data=A, kernel=(2), stride=(2), \
+                pool_type='max')
+    
+    test_pooling_large_dim()
+    A = np.ones((1, 1, 2**12, 2**10, 2**10))
```
Comment (wkcn) [inline tests/nightly/test_np_large_array.py:1016]: Can you replace the hard code number with a variable?
For example, 
```python
C, H, W = 2**12, 2**10, 2**10
A = np.ones((1, 1, C, H, W))
```

**Review by wkcn — COMMENTED** [review]
Thank you for the fix! I think some hard code should be replaced.

**Comment by Zha0q1** [comment]
> Thank you for the fix! I think some hard code should be replaced.

Sure. Thanks for reviewing

**Review by wkcn — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**